### ISSUE TITLE
Create/Share monthly Report about Contributors' TOTAL activity

### ISSUE THREAD

**Issue by aeiras** [microprofile/microprofile-marketing#27]
[stats- first graph](https://projects.eclipse.org/projects/technology.microprofile/who)

In order to be effective at welcoming new Contributors, who hopefully become at a later time Committers, we need to simply show/share Project monthly stats. 

Would like to get feedback on this before March starts. Thanks in advance MicroProfilers. 

- Thinking that (30/31 days) monthly activity as shown in the "[Who's Involved- Contributor's char](https://projects.eclipse.org/projects/technology.microprofile/who)"  under Eclipse Microprofile is a simple way to start . 

here @rstjames 

- How? twitter & any other medium used via buffer.... priority to never drop Twitter, a simply blast plus 2 mins update on the Community calls, anyone can own that update after twitter blast, scaling sharing :)  

Currently, there seems to be no way to extract reports. Will ask EF WebMaster via new issue. @eclipsewebmaster

**Comment by jroper** [comment]
Here are all contributors (with number of commits) to all `eclipse/microprofile-*` projects in the month of January:

```
njr-11: 48
Emily-Jiang: 36
rdebusscher: 28
andymc12: 17
nmittles: 15
Azquelt: 15
pilhuhn: 11
brunobat: 8
jmartisk: 8
pavolloffay: 7
doychin: 7
radcortez: 6
mmusgrov: 5
aguibert: 5
EricWittmann: 4
arthurdm: 4
thejibz: 4
phillip-kruger: 4
jmini: 4
jefrajames: 3
manovotn: 3
carlosdlr: 3
kenfinnigan: 2
xstefank: 2
cescoffier: 2
FroMage: 2
donbourne: 2
Karm: 2
Channyboy: 1
benjamin-confino: 1
aeiras: 1
raymondlam: 1
```

And for the month of February:

```
rdebusscher: 28
njr-11: 24
cescoffier: 15
jmartisk: 12
donbourne: 12
phillip-kruger: 12
mmusgrov: 8
andymc12: 6
ochaloup: 5
michalszynkiewicz: 4
Azquelt: 4
Emily-Jiang: 3
jmesnil: 2
jroper: 2
Channyboy: 2
radcortez: 2
aeiras: 2
aguibert: 2
kenfinnigan: 1
manovotn: 1
pavolloffay: 1
gsmet: 1
Ladicek: 1
carlosdlr: 1
fmhwong: 1
Karm: 1
TomasHofman: 1
```

And the code that created this summary:

https://github.com/jroper/microprofile-contributors

**Comment by aeiras** [comment]
what this is amazing @jroper! ThankYOU! 

we can already do something with Jan and reitarate. :)

**Comment by aeiras** [comment]
@jroper is it at all possible to get the stats to be automatic? 

@rstjames without rush, how likely are you to come up with a design template that is MP branded that simple enables **once** a month to share via Twitter handle the contributors' stats? 

minimum requirements, brainstorming:

- [link of stats](https://projects.eclipse.org/projects/technology.microprofile/who)

- recommended initial hashtag: #YOUhavetoSHOWup to effectively contribute to #OpenSource. 
- #MicroProfilers who rocked it in (month/year)

-  it seems not possible to write in the design all the contributors... what/how can we share that data?
maybe via taking a simple pic of there names?

**Comment by jroper** [comment]
@aeiras Depends what you mean by automatic. Where do you want the stats published?

**Comment by aeiras** [comment]
@jroper how about 1st Tuesday of every month to start this potential cool constant PR-- factual push? 
 
the thing is that the design template will be created as a contant **yet** it will need the monthly contributors' beef via the sum-activity.  

Is this something easier to achieve, delegate and make sure that more than you can help with?

I am not a fun of tasks that slave 1 person in OSS :) you know it!  Thanks James!

**Comment by rstjames** [comment]
@aeiras 
What stats do we want to show? 
Who will be supplying the stats every month?
What channels do we want to share this information on?  Only twitter seems limiting but maybe a good start...

**Comment by aeiras** [comment]
@rstjames see above original ticket I just adjusted as a follow up to you. 
Please consume ticket exchanges. 

Ryan- we are still figuring out the logistics of how to do this. 
Yet we have Jan 2018 stats ready. IS there a way to push something cool that will NOT be time consuming next time? 

- thinking for stuff that scales and that makes others with access to MP tweetdeck own

- thinking for a design template that highlights that minimum requirements for this monthly tweet.

**Comment by jroper** [comment]
@aeiras So you mean you just want a tweet posted once a month? Or do you want a web page updated? If so, which web page? And when you say design template, do you mean you want the information presented in an image?

**Comment by aeiras** [comment]
@jroper the data you provided above is perfect James. Exactly that.
The design and push logistics are irrelevant at this time. :) easier to achieve once stats are secured. 

@rstjames could provide feedback on how to present that data. thanks Ryan?

**Comment by rstjames** [comment]
@aeiras I stated above I need to know:
- what stats we want to show? We can not show everything on a tweet. What are the most important?
- is this for social media?
- is this a web page? If so this would need a meeting to discuss goals and requirements. 
- what is the goal?

**Comment by aeiras** [comment]
@rstjames this issue is messy by design. Bare with me as it has multiple components that need to be separated into new issues.

- First, update via MP social media could be achieved easier than deeper features that will come into plate [later-Social Media Projet](https://github.com/eclipse/microprofile-marketing/projects/4)

-  [what stats we want to show?   "Individual Contribution Activity" ](https://projects.eclipse.org/projects/technology.microprofile/who)
to the stats @jroper already generated Jan 2019 stats, see his comment from 4 days go.  

- i s this a web page? If so this would need a meeting to discuss goals and requirements.
There is no need for meetings. This is part of [forum thread: Updates to README files for Contributing](https://groups.google.com/forum/#!topic/microprofile/-o0Mqv4I2IY) on "

1. how easy is it to update the Contributor's git address to the click bottom in the website?

2. [drop ](https://github.com/microprofile/microprofile-site-config/blob/master/site.yaml)
 Update to new readME (once available)

Again, all collaboration is being done as as type this, so this is your space where you can direct us to see what we could be adding that is currently missing from current conversations that are a part of actually adjusting the Project page 100% in the website and only a small part of this issue.

**Comment by rdebusscher** [comment]
@aeiras Number of commits is not a good indicator of the activity of a person. Issues, issue Comments, Reviews, Forum messages, ... are not counted and are also important.

I would not use this number on Social Media.

**Comment by jroper** [comment]
We can also get information on issues, comments and reviews from GitHub, it's trivial to aggregate and summarise this data from the GitHub API, I wrote the script to get commit summaries by month in 10 minutes, I can do the same for any stat you want out of GitHub.

In theory yes, you can have people that don't commit at all but still contribute a lot to projects. My experience though working on open source projects is that at the top end of the most active contributors to a project, there tends to be a very high correlation between commits, issues, reviews, comments etc. Not 100% correlation, but high enough to say that it's not a bad indicator of the overall contributions of a person.

**Comment by aeiras** [comment]
I think we need the overall picture of contributions. 
If it is easier to pull the stats from the forum, issues, etc then great. 

We want to stay away from just commits highlights. 
This exchange is great bc this is an idea 💡 as such it requires input before we can execute it without dropping the ball. 💪🏼😋💙
